pages/newsletters/[slug].js

```diff
@@ -0,0 +1,167 @@
+import { useRouter } from 'next/router';
+import React from 'react';
+import tw from 'twin.macro';
+import Layout from '../../components/Layout.js';
+import { cachedContents } from '../../lib/cached';
+import {
+  hasuraGetNewsletter,
+  hasuraListNewsletters,
+} from '../../lib/newsletters.js';
+import { getArticleAds } from '../../lib/ads.js';
+import {
+  ArticleTitle,
+  PostTextContainer,
+  PostText,
+} from '../../components/common/CommonStyles.js';
+import {
+  hasuraLocaliseText,
+  renderNewsletterContent,
+} from '../../lib/utils.js';
+import { useAmp } from 'next/amp';
+
+const SectionContainer = tw.div`flex flex-col flex-nowrap items-center px-5 mx-auto max-w-7xl w-full`;
+
+export default function NewsletterEditionPage(props) {
+  const isAmp = useAmp();
+  const router = useRouter();
+
+  // If the page is not yet generated, this will be displayed
+  // initially until getStaticProps() finishes running
+  if (router.isFallback) {
+    return <div>Loading...</div>;
+  }
+
+  let body;
+
+  if (props.newsletter) {
+    console.log(typeof props.newsletter.content, props.newsletter.content);
+    body = renderNewsletterContent(
+      props.newsletter.content,
+      [],
+      isAmp,
+      props.siteMetadata
+    );
+  }
+
+  return (
+    <Layout
+      meta={props.siteMetadata}
+      sections={props.sections}
+      renderFooter={props.renderFooter}
+    >
+      <SectionContainer>
+        <ArticleTitle meta={props.siteMetadata} tw="text-center">
+          {props.newsletter.headline}
+        </ArticleTitle>
+        <PostText>
+          <PostTextContainer>{body}</PostTextContainer>
+        </PostText>
+      </SectionContainer>
```

Comment:
Can we add something like a standard article footer here? It would be good to have a newsletter subscribe box here especially.

Reply:
Good point, it needs something at the footer. I looked at the article footer and it links to article author and article tag indexes, which won't be available here... so I added the newsletter block at the bottom alone.